chanzuckerberg/single-cell-data-portal
# feat(cellguide): authenticated endpoint for uploads

## Reason for Change

- #6264

## Changes

- Added authenticated endpoint `/cellguide/v1/upload` 
- Added unit tests

## Testing steps

- Log in using your API key via the curator auth endpoint at `/curation/v1/auth/token`
- Add the generated token to the request header like this and run this curl request:
```
curl -X 'POST' \
  'https://pr-6289-backend.rdev.single-cell.czi.technology/cellguide/v1/upload' \
  -H 'accept: */*' \
  -H 'Authorization: Bearer PASTETOKENHERE' \
  -H 'Content-Type: application/json' \
  -d '{
  "cell_ontology_term_id": "CL_0000030",
  "description": "this is a description",
  "references": [
   "https://doi.org/10.1073/pnas.97.12.6242", "https://f1000research.com/articles/9-233/v1",
  ]
}'
```
- Should return a 201 response with the created object
- Modifying the format of `cell_ontology_term_id` should return an error
- Providing a bad link should return an error
- Missing a parameter in the request body should return an error


## Notes for Reviewer
- [this](https://github.com/chanzuckerberg/single-cell-infra/pull/958) infra PR was merged (since `rdev` couldn't upload to specified bucket)
